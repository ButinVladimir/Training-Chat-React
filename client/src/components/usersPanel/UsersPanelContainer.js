import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Socket } from 'socket.io-client';
import UsersPanel from './UsersPanel';
import { userAppears, userDisappears, updateUserList } from '../../redux/actions';
import { USER_APPEARS, USER_DISAPPEARS, UPDATE_USER_LIST } from '../../helpers/socketEvents';

const mapStateToProps = state => ({
  usersList: state.usersPanel.usersList,
});

const mapDispatchToProps = (dispatch, ownProps) => {
  const userAppearsHandler = login => dispatch(userAppears(login));
  const userDisappearsHandler = login => dispatch(userDisappears(login));
  const updateUserListHandler = users => dispatch(updateUserList(users));

  ownProps.socket.on(USER_APPEARS, userAppearsHandler);
  ownProps.socket.on(USER_DISAPPEARS, userDisappearsHandler);
  ownProps.socket.on(UPDATE_USER_LIST, updateUserListHandler);

  return {
    handlers: new Map([
      [USER_APPEARS, userAppearsHandler],
      [USER_DISAPPEARS, userDisappearsHandler],
      [UPDATE_USER_LIST, updateUserListHandler],
    ]),
  };
};

const UsersPanelContainer = connect(mapStateToProps, mapDispatchToProps)(UsersPanel);

UsersPanelContainer.propTypes = {
  socket: PropTypes.instanceOf(Socket).isRequired,
};

export default UsersPanelContainer;
