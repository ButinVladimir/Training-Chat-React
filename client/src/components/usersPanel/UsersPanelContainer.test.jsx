import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import UsersPanelContainer from './UsersPanelContainer';

describe('UsersPanelContainer', () => {
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore();
  });

  it('renders without crashing', () => {
    const store = mockStore({
      usersPanel: {
        usersList: ['user1', 'user2', 'user3'],
      },
    });
    const component = mount(
      <Provider store={store}>
        <UsersPanelContainer />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
