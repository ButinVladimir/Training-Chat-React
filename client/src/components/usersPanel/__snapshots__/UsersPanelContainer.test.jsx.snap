// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersPanelContainer renders without crashing 1`] = `
<div
  className="chat-users-list-container"
>
  <ul
    className="users-list"
  >
    <li
      className="user-line"
      key="user1"
    >
      <button
        className="user-select"
        onClick={[Function]}
        type="button"
      >
        user1
      </button>
    </li>
    <li
      className="user-line"
      key="user2"
    >
      <button
        className="user-select"
        onClick={[Function]}
        type="button"
      >
        user2
      </button>
    </li>
    <li
      className="user-line"
      key="user3"
    >
      <button
        className="user-select"
        onClick={[Function]}
        type="button"
      >
        user3
      </button>
    </li>
  </ul>
</div>
`;
